processing/src/main/java/io/druid/segment/filter/LikeFilter.java

```diff
@@ -156,27 +86,59 @@ public boolean supportsSelectivityEstimation(
     return Filters.supportsSelectivityEstimation(this, dimension, columnSelector, indexSelector);
   }
 
-  private static Iterable<ImmutableBitmap> getBitmapIterator(
-      final BitmapIndex bitmapIndex,
-      final LikeDimFilter.LikeMatcher likeMatcher,
-      final Indexed<String> dimValues
-  )
+  private Iterable<ImmutableBitmap> getBitmapIterable(final BitmapIndexSelector selector)
+  {
+    if (isSimpleEquals()) {
+      // Verify that dimension equals prefix.
+      return ImmutableList.of(selector.getBitmapIndex(dimension, likeMatcher.getPrefix()));
+    } else if (isSimplePrefix()) {
+      // Verify that dimension startsWith prefix, and is accepted by likeMatcher.matchesSuffixOnly.
+      final BitmapIndex bitmapIndex = selector.getBitmapIndex(dimension);
+
+      if (bitmapIndex == null) {
+        // Treat this as a column full of nulls
+        return ImmutableList.of(likeMatcher.matches(null) ? Filters.allTrue(selector) : Filters.allFalse(selector));
+      }
+
+      // search for start, end indexes in the bitmaps; then include all matching bitmaps between those points
+      final Indexed<String> dimValues = selector.getDimensionValues(dimension);
+
+      // Union bitmaps for all matching dimension values in range.
+      // Use lazy iterator to allow unioning bitmaps one by one and avoid materializing all of them at once.
+      return Filters.bitmapsFromIndexes(
+          getDimValueIndexIterableForPrefixMatch(bitmapIndex, dimValues),
+          bitmapIndex
+      );
+    } else {
+      // fallback
+      return ImmutableList.of(
+          Filters.matchPredicate(
```

Comment:
When `getBitmapIterable()` is called from `estimateSelectivity()`, such a fallback is nonsense, because the whole point of `estimateSelectivity()` is to avoid materializing _this_ bitmap. It could be fixed by adding a method like `Iterable<ImmutableBitmap> Filters.predicateMatchingBitmaps()`.